src/pal/src/arch/arm/context2.S

```diff
@@ -192,13 +192,30 @@ LEAF_ENTRY RtlRestoreContext, _TEXT
     ldr R2, [r0, #(CONTEXT_Cpsr)]
     msr APSR, r2    
     
-    mov r12, r0 // ideally we would ldmia r0, {r0-r12, sp, lr, pc} here, but that isn't supported on new clang
-    add r12, CONTEXT_R0 // so we'll burn r12 as the IPC reg for now -- TODO: is this ok?
-    ldm r12, {r0-r11}
-    add r12, (CONTEXT_Sp - CONTEXT_R0)
+    // Ideally, we would like to use `ldmia r0, {r0-r12, sp, lr, pc}` here,
+    // but clang 3.6 and later, as per ARM recommendation, disallow using
+    // Sp in the register list, and Pc and Lr simmultaneously. So we restore
+    // Sp first, then Lr. Then we copy the content of Pc to where Sp was stored
+    // in Context argument before issuing the ldm instruction to restore all
+    // r0 to r12 registers and Pc.
+    // This is ok to overwrite the existing value since the context argument is
+    // not used anymore after we jump to the new Pc.
+
+    // Move `r0' to the r0 location in context argument.
+    add r0, CONTEXT_R0
+    mov r12, r0
+    // Update r12 to the Sp location in context argument
+    // and restore Sp and Lr.
+    add r12, (CONTEXT_Sp - (CONTEXT_R0))
     ldr sp, [r12]
-    ldr lr, [r12, #(CONTEXT_Lr - CONTEXT_Sp)]
-    ldr pc, [r12, #(CONTEXT_Pc - CONTEXT_Sp)]
+    ldr lr, [r12, #(CONTEXT_Lr - (CONTEXT_Sp))]
```

Comment:
I thought about not restoring r0-r3 too. Is there a point in storing them in RtlCaptureContext? As for lr we need to restore it as it is used to return to the caller.